gradio/helpers.py

```diff
@@ -690,6 +690,26 @@ def special_args(
         elif type_hint == routes.Request:
             if inputs is not None:
                 inputs.insert(i, request)
+        elif (
+            type_hint == Optional[oauth.OAuthProfile] or type_hint == oauth.OAuthProfile
+        ):
+            if inputs is not None:
+                # Retrieve session from gr.Request, if it exists (i.e. if user is logged in)
+                session = (
+                    # request.session (if fastapi.Request obj i.e. direct call)
+                    getattr(request, "session", {})
+                    or
+                    # or request.request.session (if gr.Request obj i.e. websocket call)
+                    getattr(getattr(request, "request", None), "session", {})
+                )
+                oauth_profile = (
+                    session["oauth_profile"] if "oauth_profile" in session else None
+                )
+                if type_hint == oauth.OAuthProfile and oauth_profile is None:
+                    raise ValueError(
```

Comment:
You can raise a `gr.Error()` instead and it should propagate to the frontend

Reply:
Great! That's what I needed :)

Reply:
Added in https://github.com/gradio-app/gradio/pull/4943/commits/103fba5ad947f14870b2a0934f5701c0625740eb